### PR TITLE
add noto fonts recipe

### PR DESCRIPTION
**Pull request by izahn**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by izahn** [comment]
This PR is ready for review @conda-forge/staged-recipes 

There are a couple of things I'm not sure about:

1. Google doesn't version the full noto-fonts collection as far as I can tell. I used the data of the last update announcement as the version number, but am open to other suggestions.
2. The package is large, almost 990MB! The good news is that it looks like it is updated very infrequently and is noarch, but this size gives me some pause.

Let me know what you think, or if there are any other issues with this recipe.

File: recipes/font-noto/meta.yaml
Comment (chrisburr) [inline recipes/font-noto/meta.yaml:8]: I haven't checked if it's suitable but is it possible to use the versioned URL on GitHub: https://github.com/googlefonts/noto-fonts/releases/tag/v20201206-phase3

(it also seems to be a newer version?)

Comment (izahn) [reply recipes/font-noto/meta.yaml:8]: That one excludes CJK which is really what I need noto for in the first place. It might be better to package https://github.com/googlefonts/noto-fonts and https://github.com/googlefonts/noto-cjk separately, but on the other hand I like the simplicity of just packaging the whole thing. Let me know what you think, I'm happy to split them out if that makes more sense.

**Comment by izahn** [comment]
Any further suggestions here? I think keeping this together in one monolithic package is simplest, but happy to split it out into `noto-font` and `noto-font-cjk` if needed.

**Comment by isuruf** [comment]
For what do you want this? Why is it so large?

**Comment by izahn** [comment]
I want it to view Chinese text. It's big because you need a lot of glyphs to avoid tofu in general

**Comment by izahn** [comment]
@conda-forge/staged-recipes is there anything further I can do here? If this is not a good fit for conda-forge please let me know and I'll close this PR and host it in my personal channel. Otherwise it would be great to get this merged. Thank you for your efforts, I truly appreciate all the help and support.

**Comment by carterbox** [comment]
@conda-forge/core, is a font considered software in the context of conda-forge?

**Comment by carterbox** [comment]
https://github.com/conda-forge/font-ttf-inconsolata-feedstock
https://github.com/conda-forge/font-ttf-dejavu-sans-mono-feedstock

Looks like we already have a few fonts published on conda-forge

File: recipes/font-noto/meta.yaml
Comment (carterbox) [reply recipes/font-noto/meta.yaml:8]: Please use the versioned releases from the googlefonts github org and have separate recipes for each font group: noto, cjk, and emoji. You can just add additional folders to this PR, so each of the packages can be reviewed simultaneously.

**Review by carterbox — CHANGES_REQUESTED** [review]
@izahn, I'd be happy to merge this for you pending changes unless someone from core overrules me before it's merged.

In addition to splitting this package into the smaller versioned sections, please change the name to match the other font packages (prefix with "font-ttf").

**Comment by isuruf** [comment]
The issue is not that the the package is a font. It's that it's so large, this is going to take up 1GB of space per package and I'm not sure that's a good idea or not.

**Comment by izahn** [comment]
I'm going to split this up and package cjk separately. That will cut down the size while still providing good coverage. Will try to get to this over the weekend.

**Comment by carterbox** [comment]
cudatoolkit tarballs are also around 1GB. 🤷🏼

File: recipes/font-noto/meta.yaml
Comment (chrisburr) [reply recipes/font-noto/meta.yaml:8]: > separate recipes for each font group: noto, cjk, and emoji

I wonder if it's worth splitting even finer than that, perhaps on a [per folder/prefix basis](https://github.com/googlefonts/noto-fonts/tree/main/hinted/ttf) similar to [how fedora handles this](https://koji.fedoraproject.org/koji/buildinfo?buildID=1930689). There can still be a meta-package for convienence to get the entire set.

This would also match with how the fonts are being [versioned in practice](https://github.com/googlefonts/noto-fonts#recent-updates) so it would also help minimise the volume of data generated by updates.

Comment (izahn) [reply recipes/font-noto/meta.yaml:8]: Usually conda-forge doesn't split things up that small, e.g., fedora packages each `qt5` module separately, 40 in all, while conda-forge puts everything in one `qt` package.

I'm going to start with `noto / noto-cjk / noto-emoji` and see what it looks like.

Comment (hmaarrfk) [reply recipes/font-noto/meta.yaml:8]: > Usually conda-forge doesn't split things up that small, e.g., fedora packages each qt5 module separately, 40 in all, while conda-forge puts everything in one qt package.

I think this approach is slowly falling out of favor as people are starting to integrate more and more things together. For one, I am very interested in splitting up qt into smaller packages, but it isn't so trivial.

Comment (izahn) [reply recipes/font-noto/meta.yaml:8]: Sure, qt was a bad example maybe. But I don't think we should aspire to split things up as much as Fedora does either.

**Comment by izahn** [comment]
> @izahn, I'd be happy to merge this for you pending changes unless someone from core overrules me before it's merged. But, since we already distribute fonts on conda-forge, I'm not sure what the case against would be.
> 
> In addition to splitting this package into the smaller versioned sections, please change the name to match the other font packages (prefix with "font-ttf").

Done in https://github.com/conda-forge/staged-recipes/pull/16032/commits/bb8ac9157976b38c0ba07f7292e28a408c7c11be

**Comment by izahn** [comment]
Hmmm, getting 
```
ERROR:/home/conda/staged-recipes-copy/.ci_support/compute_build_graph.py:invalid recipe dir: /home/conda/staged-recipes-copy/recipes/font-ttf-noto
```
Does anyone know what this means?

File: recipes/font-ttf-noto/build.sh
Comment (carterbox) [inline recipes/font-ttf-noto/build.sh:4]: So the actual conda-forge package size is much smaller than 1GB upstream source release because you are only including TTF files from this single directory? Can you comment why you have chosen the unhinted variable TTF fonts instead of the OTF fonts or the hinted TTF fonts?

**Comment by carterbox** [comment]
I removed the "needs decision" label because upon further investigation, it seems like the concern over large packages sizes is not necessary because there is a lot of redundant information the source releases from upstream. Firstly, the fonts are provided in multiple file formats, but I think our recipe will just pick one of these fonts to distribute. Secondly, the main font source contains a folder called "archive" which basically doubles the size of the download. My understanding is that these are just old versions of fonts and do not need to be included in our distribution.

File: recipes/font-ttf-noto/build.sh
Comment (izahn) [reply recipes/font-ttf-noto/build.sh:4]: Yes, I think we can get the size down. I chose .ttf because that's what other font package in conda-forge seem to provide. I chose unhinted because this is the only format that provides variable fonts as far as I can tell.

**Comment by izahn** [comment]
Yes, I think we can get the size down and still meet my goal of comprehensive coverage.

The problem right now is that the CJK fonts archive has to issue that prevents it from being extracted. The error says 
```
libarchive.exception.ArchiveError: Path contains '..' (errno=-1, retcode=-25, archive_p=94306351011216)
```
I plan to report this upstream and see if we can get them to fix it for us.

**Comment by izahn** [comment]
I have opened an issue upstream in https://github.com/googlefonts/noto-cjk/issues/216

**Comment by izahn** [comment]
@conda-forge-admin please restart ci

**Comment by izahn** [comment]
> I have opened an issue upstream in [googlefonts/noto-cjk#216](https://github.com/googlefonts/noto-cjk/issues/216)

This has been fixed upstream.

**Comment by izahn** [comment]
I think this is ready for another round of review. I've split the package into three and got the size down to
- font-ttf-noto: 22M
- font-ttf-noto-cjk: 180M
- font-ttf-noto-emoji: 19M

**Review by carterbox — CHANGES_REQUESTED** [review]
Since you're not including all of the possible formats of the fonts in the package, could you please include in the summary for each package a description of which fonts you are including. i.e. hinted/unhinted variable/nonvariable. I'm not sure what exactly you excluded included because I don't know anything about fonts.

Also, some of the package summaries are just wrong. The noto package has the description for the CJK fonts.

I'm thinking something like:

"Google's Noto font in unhinted variable TTF format. Excludes CJK characters and emojii."

"Google's Noto emojii in TTF format."

"Google's Noto font for CJK characters in fixed and variable width TTF format."

**Comment by izahn** [comment]
Oops, I did mix up the noto and noto-cjk summaries, thanks for catching that!

I expanded the summaries to note where only variable fonts were included. I didn't include TTF in the description, since that is in the package names already. I also didn't mention google ( I don't think we generally mention the copy-write holder in the description). Let me know if you really do want those elements included, I just left them out to be more consistent with other conda-forge package sumaries.

**Review by carterbox — APPROVED** [review]

**Pull request merged**
